src/org/opendatakit/briefcase/ui/reused/source/ShowSourceForm.java

```diff
@@ -32,26 +32,44 @@
   private JLabel sourceLabel;
   private JButton resetButton;
   private JLabel actionLabel;
+  private JButton reloadButton;
   private List<Runnable> onResetCallbacks = new ArrayList<>();
+  private List<Runnable> onReloadCallbacks = new ArrayList<>();
+  private Operation type;
 
-  private ShowSourceForm(String action) {
+  private ShowSourceForm(String action, Operation type) {
     $$$setupUI$$$();
     this.action = action;
+    this.type = type;
 
     resetButton.addActionListener(__ -> onResetCallbacks.forEach(Runnable::run));
+    if (type.equals(Operation.PULL)) {
+      reloadButton.addActionListener(__ -> onReloadCallbacks.forEach(Runnable::run));
+    } else {
+      reloadButton.setVisible(false);
+    }
   }
 
-  static ShowSourceForm empty(String action) {
-    return new ShowSourceForm(action);
+  static ShowSourceForm pull(String action) {
```

Comment:
I'm a huge fan of these specialized factories. I think they're neat and that they add super useful semantics.

Reply:
Than you :smile:. I took inspiration from the similar factories elsewhere in the nearby code base.